VectorInstitute/FLorist
# Pick a database to store training results

# PR Type
Feature

# Short Description

Clickup Ticket(s): https://app.clickup.com/t/8687c8g5x

In short, here I'm seeting up MongoDB as the database of choice and also creating a new database entity and API route just to startup and test the database.

In details, I am:
- Adding instructions on how to set up MongoDB
- Adding `pymongo` and `pydantic` to our dependencies
- Adding MongoDB setup code to startup and shutdown of server's FastAPI
- Adding MongoDB github action to the integration test setup
- Adding boilerplate code to setup MongoDB in integration tests
- Adding a basic `Job` entity to store information about training jobs. **NOTE**: this is not final and will change a lot over time as we add more and more functionality.
- Adding an endpoint to create new training jobs
- Adding an integration test to that endpoint

If you want to test this yourself:
1. Setup MongoDB on your local machine as directed in the README
1. Start the server with `yarn dev`
1. On postman, set up a post request to `localhost:8000/api/server/job` and add the following to the request body as a raw json:
```json
{
  "model": "MNIST",
  "redis_host": "locahost",
  "redis_port": "6879"
}
``` 
You should receive a response with the new record created in the database.

# Tests Added
`florist/tests/integration/api/routes/server/test_job.py`
